### ISSUE TITLE
error: expected function body after function declarator AVAILABLE_MAC_OS_X_VERSION_10_10_AND_LATER in IOHIDDevice.h

### ISSUE THREAD

**Issue by matthewbauer** [NixOS/nixpkgs#13779]
## Basic info

To make sure that we are on the same page:

* Kernel: (run `uname -a`)
Darwin arda 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64

* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.1

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
16.09pre78318.e82dcd7

## Describe your issue here
It looks like IOKit 10.10 needs a libSystem that is also 10.10 (currently libSystem and the rest of stdenv is still at 10.9). A quick fix could patch this line up but I'm thinking that more problems are going to come. Is there a reason we haven't update libSystem to 10.10/10.11? I've looked into and it looks like there a lot of patches that will have to be redone because enough stuff is thrown off.

```
In file included from /nix/store/dfvrzc0p99vp3gx2q7w79iiq14drj851-IOKit-osx-10.10.5/Library/Frameworks/IOKit.framework/Headers/hid/IOHIDLib.h:31:
/nix/store/dfvrzc0p99vp3gx2q7w79iiq14drj851-IOKit-osx-10.10.5/Library/Frameworks/IOKit.framework/Headers/hid/IOHIDDevice.h:315:1: error: expected function body after function declarator
AVAILABLE_MAC_OS_X_VERSION_10_10_AND_LATER;
^
```

### Steps to reproduce

```sh
git clone https://github.com/NixOS/nixpkgs.git
cd nixpkgs
git fetch https://github.com/matthewbauer/nixpkgs.git darwin-fixes
git checkout 76748c3aa6d9cf4c4911186e647505f50746f840
nix-build -A mplayer
```

**Comment by copumpkin** [comment]
Main reason is that it wasn't trivial when I tried bumping the versions, so I made them inconsistent instead. I'd love for someone to take care of bumping the remaining ones and ironing out issues that arise. It can be a bit of a delicate dance though, given the relationship with the bootstrap tools.

**Comment by ryanartecona** [comment]
After starting down the path to fix mplayer on darwin myself, I also hit this.

@copumpkin It's likely I won't be able to help much, but if I wanted to take a stab at doing the Libsystem et al. version bump, where would I start? Can you give a quick lay of the land of darwin stdenv and bootstrap stuff, or point me to where I could read up on it (hoping beyond hope that docs exist for it at all)?

**Comment by matthewbauer** [comment]
You can actually build mplayer if you change 10_11 to 10_10.

**Comment by ChrisJefferson** [comment]
Could we, as a temporary fix, just build everything with a 10.9 set of SDKs, rather than updating some to 10.11 and leaving others at 10.9?

**Comment by matthewbauer** [comment]
@ChrisJefferson I suppose so though that could break as many or more things depending on what packages are actually using. Probably safer to just get everything on the same page.
